### DISCUSSION TITLE
TheXTech v1.3.6.1: Internationalisation, OpenGL, etc.

### DISCUSSION THREAD

**Discussion by Wohlstand** [Wohlstand/TheXTech#625] [General]
We worked on this update long and hard, and now we're ready to present a bunch of new features and fixes! This is a checkpoint release after which there are major works will be started. But don't worry, if any issue occurs with this version, hotfixes will be released.

This is the first version that introduces true kind of internationalisation: now it's possible to have game being translated into many languages, including translations for episodes! (to translate episodes, the [not-yet-released Moondust Translator](https://github.com/WohlSoft/Moondust-Project/tree/wip-translator) component is required).

This update introduces the experimental OpenGL rendering that brings back the bitmask render (which is required to accurately reproduce the glitchy colours rendering of some levels that do use bitmask render for tricks).

**Note:** The rest of assets packages had been updated to add new directories `fonts` and `languages` that contains new fonts that allows to print non-ASCII characters, and some translations for the engine (currently, Russian and Chinese [both Simplified and Traditional] translations are available at this release, and other incomplete translations are available at the official repository). You may want to [contribute to the translation process using the WebLate platform](https://hosted.weblate.org/projects/thextech/). A new file called `editor.ini` has also been added to enable customization of the in-game editor's features and is now required to use the editor.

# Most important changes
- **Added real kind of internationalisation: fonts are able to print non-ASCII, TTF support, and multi-languages support.** The game since now can be translated into other languages. This update makes TheXTech being a first branch of SMBX that features native multi-language support. While SMBX being English-only all the time, gives a lot of limitations to the whole world: a lot of people just don't know English, and the multi-languages support will get them to better understand the game and much easier get into it!
- **Experimental OpenGL support** that allows bitmask render to bring back the support for colour tricks that were used at several vanilla levels and episodes and packed games (for example, "Reign of the Gatekeepers").
- **Added Homebrew Switch support.** The game is now able to work on hacked Switch consoles or on emulators such as Yuzu.
- **Updated SDL2 and MixerX libraries.** There are more game controllers got supported, and the support of existing controllers had been fixed somewhere. Now the game supports new music formats such as WavPack (wv) and PXTone (ptcop and pttune).
- **Added opt-in Discord Rich Presense support** for these players who want to tell more about the game that they play. It's disabled by default, and you can enable it by editing of the `settings/thextech.ini` file.
- **The keyhole framerate fix now enabled permanently.** That means, if you run an episode for speedrun, pay attention that keyhole exit will add 192 ticks instead of 300 as it was originally. The vanilla game used a different frame rate (100fps instead of 65fps) during keyhole exits which led to an inflated In-Game Time.
- **Various in-game editor features** including the Magic Block automatic tiling mode, a first for the SMBX ecosystem!
- **Performance improvements** that lower the game logic's CPU consumption on low-end systems.

Read full changelog at the "changelog.txt" file in the packages or in the source code archives.

# Screenshots
![Scr_2023-09-03_16-57-32](https://github.com/Wohlstand/TheXTech/assets/6751442/2493054a-8636-45ab-80fa-f425e16464bf)

![Scr_2023-09-03_18-29-27](https://github.com/Wohlstand/TheXTech/assets/6751442/323b7823-8606-4a2d-8ea7-f6cfce5e143f)

![Scr_2023-09-03_18-31-10](https://github.com/Wohlstand/TheXTech/assets/6751442/539494a6-f8da-4d70-8e19-0d28f9446337)

![Scr_2023-09-03_18-31-58](https://github.com/Wohlstand/TheXTech/assets/6751442/6e3a388e-9c63-4b67-9aed-07434f24a060)

![Scr_2023-09-03_18-36-05](https://github.com/Wohlstand/TheXTech/assets/6751442/afdd0042-5482-4da7-870f-92e28b0e59d8)

![Scr_2023-09-03_18-41-43](https://github.com/Wohlstand/TheXTech/assets/6751442/e387ee9f-a67b-4857-b640-5234d652c4de)

<hr /><em>This discussion was created from the release <a href='https://github.com/Wohlstand/TheXTech/releases/tag/v1.3.6.1'>TheXTech v1.3.6.1: Internationalisation, OpenGL, etc.</a>.</em>

**Comment by P3rf3ctXZer0** [comment]
Congrats on this huge milestone!

**Reply by Wohlstand** [reply]
Thanks :)

**Comment by LoveBodhi** [comment]
暖暖。
